x-pack/filebeat/tests/integration/input_reload_test.go

```diff
@@ -0,0 +1,269 @@
+					Id:   "log-input",
+					Type: "log",
+					Name: "log",
+					Streams: []*proto.Stream{
+						{
+							Id: "log-input-1",
+							Source: requireNewStruct(t, map[string]interface{}{
+								"enabled": true,
+								"type":    "log",
+								"paths":   []interface{}{logFilePath},
+							}),
+						},
+					},
+				},
+			},
+		},
+		{
+			{
+				Id:             "output-unit",
+				Type:           proto.UnitType_OUTPUT,
+				ConfigStateIdx: 1,
+				State:          proto.State_HEALTHY,
+				LogLevel:       proto.UnitLogLevel_DEBUG,
+				Config: &proto.UnitExpectedConfig{
+					Id:   "default",
+					Type: "elasticsearch",
+					Name: "elasticsearch",
+					Source: requireNewStruct(t,
+						map[string]interface{}{
+							"type":     "elasticsearch",
+							"hosts":    []interface{}{"http://localhost:9200"},
+							"username": "admin",
+							"password": "testing",
+							"protocol": "http",
+							"enabled":  true,
+						}),
+				},
+			},
+			{
+				Id:             "input-unit-2",
+				Type:           proto.UnitType_INPUT,
+				ConfigStateIdx: 1,
+				State:          proto.State_HEALTHY,
+				LogLevel:       proto.UnitLogLevel_DEBUG,
+				Config: &proto.UnitExpectedConfig{
+					Id:   "log-input",
+					Type: "log",
+					Name: "log",
+					Streams: []*proto.Stream{
+						{
+							Id: "log-input-2",
+							Source: requireNewStruct(t, map[string]interface{}{
+								"enabled": true,
+								"type":    "log",
+								"paths":   []interface{}{logFilePath},
+							}),
+						},
+					},
+				},
+			},
+		},
+	}
+
+	idx := 0
+	waiting := false
+	when := time.Now()
+	nextState := func() {
+		if waiting {
+			if time.Now().After(when) {
+				idx = (idx + 1) % len(units)
+				waiting = false
+				return
+			}
+			return
+		}
+		waiting = true
+		when = time.Now().Add(10 * time.Second)
+	}
+	server := &mock.StubServerV2{
+		CheckinV2Impl: func(observed *proto.CheckinObserved) *proto.CheckinExpected {
+			if management.DoesStateMatch(observed, units[idx], 0) {
+				nextState()
+			}
+			for _, unit := range observed.GetUnits() {
+				if state := unit.GetState(); !(state == proto.State_HEALTHY || state != proto.State_CONFIGURING || state == proto.State_STARTING) {
+					t.Fatalf("Unit '%s' is not healthy, state: %s", unit.GetId(), unit.GetState().String())
+				}
+			}
+			return &proto.CheckinExpected{
+				Units: units[idx],
+			}
+		},
+		ActionImpl: func(response *proto.ActionResponse) error {
+			return nil
+		},
+		ActionsChan: make(chan *mock.PerformAction, 100),
+	}
+
+	require.NoError(t, server.Start())
+	defer server.Stop()
+
+	p := NewBeat(
+		t,
+		"../../filebeat.test",
+		[]string{
+			"-E", fmt.Sprintf("management.insecure_grpc_url_for_testing=\"localhost:%d\"", server.Port),
+			"-E", "management.enabled=true",
+		},
+		tempDir,
+	)
+
+	p.Start()
+
+	// logger: centralmgmt
+	// logger: centralmgmt.V2-manager
+	// logger: centralmgmt.V2-manager
+	// logger: centralmgmt.V2-manager
+	// logger: centralmgmt.V2-manager
```

Comment:
Are those needed?

Reply:
I'll clean this up once CI is happy. Those are the loggers of every log line I'm looking for.